tests/integ/scala/test_snowflake_plan_suite.py

```diff
@@ -250,15 +263,21 @@ def test_create_scoped_temp_table(session):
             == f' CREATE  TEMPORARY  TABLE {temp_table_name}("NUM" BIGINT, "STR" STRING(8))'
         )
         assert (
-            session._plan_builder.create_temp_table(
-                temp_table_name,
+            session._plan_builder.save_as_table(
+                [temp_table_name],
+                None,
+                SaveMode.APPEND,
+                "temp",
+                None,
+                None,
                 df._plan,
+                None,
                 use_scoped_temp_objects=True,
                 is_generated=False,
             )
             .queries[0]
             .sql
-            == f' CREATE  TEMPORARY  TABLE {temp_table_name}("NUM" BIGINT, "STR" STRING(8))'
+            == f' CREATE  TEMPORARY  TABLE {temp_table_name} If  NOT  EXISTS ("NUM" BIGINT, "STR" STRING(8))'
```

Comment:
if we do no go through append, do we still have this issue here

Reply:
if we do not go through append, we will not have a create temp table statement. instead we will have CTAS statement.